receiver/receivercreator/README.md

```diff
@@ -438,3 +438,186 @@ service:
 
 The full list of settings exposed for this receiver are documented [here](./config.go)
 with detailed sample configurations [here](./testdata/config.yaml).
+
+
+## Generate receiver configurations from provided Hints
+
+Currently this feature is only supported for K8s environments and the `k8sobserver`.
+
+The feature for K8s is enabled with the following setting:
+
+```yaml
+receiver_creator/metrics:
+  watch_observers: [ k8s_observer ]
+  hints:
+    k8s:
+      enabled: true
+```
+
+Users can use the following annotations to automatically enable receivers to start collecting signals from the target Pods/containers.
+
+### Supported metrics annotations
+
+#### Enable/disable discovery
+
+`opentelemetry.io.discovery/enabled` (example: `"true"`)
+
+By default `"true"`.
```

Comment:
I think this can be a bit misleading with the current logic's; what will happen if a user edits an annotation from `true` to `false` after the receiver creator has already created the hints' receiver? I would expect the receiver_creator to shut down the created receiver. 

Likewise, I am not sure if the receiver_creator is able to perform the latest, what about if for the moment we remove `opentelemetry.io.discovery/enabled` and assume that if the other fields are defined the user expects the receiver to be created?

Reply:
> I would expect the receiver_creator to shut down the created receiver

I believe that's exactly what will happen. If Pod's annotations change then [`OnChange`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/fa20d6a9e8c34be8fef2de806a2a7350b3c62c32/receiver/receivercreator/observerhandler.go#L150) is called  which stops+starts the receiver(s). However if the annotation `opentelemetry.io.discovery/enabled: "false"` is added the receivers won't start again because `start` will take through the whole annotations' evaluation again and will take into account the newly added annotation.

Reply:
My assumptions were wrong then, thanks for the detailed explanation and testing it. LGTM